File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/BastardInjection/_4_BastardInjection/Services/MsSqlBasedRepository.java
using BastardInjection._4_BastardInjection.Interfaces;

namespace BastardInjection._4_BastardInjection.Services
{
  public interface IRepository
  {
    void Save(AcmeMessage message);
  }

  public class MsSqlBasedRepository : IRepository
  {
    private readonly DataDestination _dataDestination;

    public MsSqlBasedRepository() : this(new SqlDataDestination())
    {
      
    }

    //for tests
    public MsSqlBasedRepository(DataDestination dataDestination)
    {
      _dataDestination = dataDestination;
    }

    public void Save(AcmeMessage message)
    {
      message.WriteTo(_dataDestination);
    }
  }
}

File: CSharp/DomainSpecificLanguages/DomainSpecificLanguages/AlarmsNestedFunctions/Services/MsSqlBasedRepository.cs
using TelecomSystemNestedFunctions.Interfaces;

namespace TelecomSystemNestedFunctions.Services
{
  public interface IRepository
  {
    void Save(AcmeMessage message);
  }

  public class MsSqlBasedRepository : IRepository
  {
    private readonly DataDestination _sqlDataDestination;

    public MsSqlBasedRepository(DataDestination sqlDataDestination)
    {
      _sqlDataDestination = sqlDataDestination;
    }

    public void Save(AcmeMessage message)
    {
      message.WriteTo(_sqlDataDestination);
    }
  }
}